File: Java/Desktop/NewLab8/src/by/bntu/fitr/povt/cng/model/entity/container/SmartTree.java
package by.bntu.fitr.povt.cng.model.entity.container;

import by.bntu.fitr.povt.cng.model.collections.deprecated.CustomCollections;
import by.bntu.fitr.povt.cng.model.entity.abstracts.Toy;

public class SmartTree {

    private CustomCollections <Toy> toysMass = null;
    private static SmartTree tree = null;

    public SmartTree() {

    }

    public SmartTree(CustomCollections collections){
        toysMass = collections;
    }

    public static SmartTree getInstance(CustomCollections collections){
        return new SmartTree(collections);
    }

    public Toy[] getToysMass() {
        return (Toy[]) toysMass.CollectionToArray();
    }

    public int getSize(){
        return toysMass.size();
    }

    public Toy getToy(int index){
        return toysMass.get(index);
    }
    //priceTree
    public void changeToy(Toy newToy, int index){
        toysMass.setElement(index,newToy);
    }

    public void setToy(Toy newToy, int index){
        toysMass.setElement(index,newToy);
    }

    public double getPrice(int index){
        return toysMass.get(index).getPrice();
    }

    public void add(Toy...newToys) {
        for (Toy newToy : newToys){
            toysMass.add(newToy);
        }
    }

    @Override
    public int hashCode() {
        int HASHCONST = 31;
        int result = 1;
        result = HASHCONST * result + getSize();
        for (Object object : toysMass.CollectionToArray()){
            result = HASHCONST * result + object.hashCode();
        }
        return result;
    }

    @Override
    public boolean equals(Object obj) {
        return (obj != null && this.hashCode() == obj.hashCode() && obj.getClass() == this.getClass());
    }

    @Override
    public String toString() {
        StringBuilder sb = new StringBuilder();
        for (int i = 0; i < toysMass.size(); i++) {
            sb.append(toysMass.get(i)).append("\n");
        }
        return sb.toString();
    }

}

File: Csharp/Lab8/Lab8/Lab8/cng/Model/Entity/Container/SmartTree.cs
using System;
using System.Text;
using Lab8.cng.Model.Collections.deprecated;
using Lab8.cng.Model.Entity.Abstracts;

namespace Lab8.cng.Model.Entity.Container
{
    public class SmartTree
    {
        private ICustomCollections <Toy> _toysMass = null;
        private static SmartTree _tree = null;

        public SmartTree() {

        }

        public SmartTree(ICustomCollections<Toy> collections){
            _toysMass = collections;
        }

        public static SmartTree GetInstance(ICustomCollections<Toy> collections){
            return new SmartTree(collections);
        }


        public Toy[] GetToysMass() {    
            return (Toy[]) _toysMass.CollectionToArray();
        }

        public int GetSize(){
            return _toysMass.Size();
        }

        public Toy GetToy(int index){
            return _toysMass.Get(index);
        }
        //priceTree
        public void ChangeToy(Toy newToy, int index){
            _toysMass.SetElement(index,newToy);
        }

        public void SetToy(Toy newToy, int index){
            _toysMass.SetElement(index,newToy);
        }

        public double GetPrice(int index){
            return _toysMass.Get(index).GetPrice();
        }

        public void Add(Toy []newToys) {
            foreach (Toy newToy in newToys){
                _toysMass.Add(newToy);
            }
        }

        
        public override int GetHashCode() {
            int hashconst = 31;
            int result = 1;
            result = hashconst * result + GetSize();
            foreach (Object obj in _toysMass.CollectionToArray()){
                result = hashconst * result + obj.GetHashCode();
            }
            return result;
        }

        
        public bool Equals(Object obj) {
            return (obj != null && this.GetHashCode() == obj.GetHashCode() && obj.GetType() == this.GetType());
        }

        
        public override String ToString() {
            StringBuilder sb = new StringBuilder();
            for (int i = 0; i < _toysMass.Size(); i++) {
                sb.Append(_toysMass.Get(i)).Append("\n");
            }
            return sb.ToString();
        }
    }
}